Guide/Constructive/Bonetrousle/README.md

```diff
@@ -0,0 +1,97 @@
+# Bonetrousle
+[https://www.hackerrank.com/challenges/bonetrousle](https://www.hackerrank.com/challenges/bonetrousle)
+
+Difficulty: Medium
+
+
+## Problem
+### Statement
+Once upon a time, Papyrus the skeleton went to buy some pasta from the store. The store's inventory is bare-bones and they only sell one thing — boxes of uncooked spaghetti! The store always stocks exactly k boxes of pasta, and each box is numbered sequentially from 1 to k. This box number also corresponds to the number of sticks of spaghetti in the box, meaning the first box contains 1 stick, the second box contains 2 sticks, the third box contains 3 sticks, …, and the k<sup>th</sup> box contains k sticks. Because they only stock one box of each kind, the store has a tendon-cy to sell out of spaghetti.
```

Comment:
This is directly copied from Hackerrank, according to their user agreement we are not allowed to do that. Please summarize the essence of the problem, the key ideas involved, rather than copying the problem statement.